src/main/java/EchoServer.java

```diff
@@ -0,0 +1,12 @@
+import interfaces.IServerSocket;
+
+public class EchoServer {
+
+    public EchoServer(IServerSocket serverSocketWrapper) {
```

Comment:
**praise**: I like that you are thinking in terms of "wrapping".

**suggestion**: You are depending on the `IServerSocket` type, but your parameter name suggests a concrete type. I would recommend naming the parameter to reflect the abstraction/interface instead of something that implements the interface.